sdk/node/test/accounts.js

```diff
@@ -0,0 +1,246 @@
+/* eslint-env mocha */
+
+const chain = require('../dist/index.js')
+const uuid = require('uuid')
+const assert = require('assert')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+
+chai.use(chaiAsPromised)
+const expect = chai.expect
+
+const client = new chain.Client()
+const signer = new chain.HsmSigner()
+
+const aliceAlias = `alice-${uuid.v4()}`
+const bobAlias = `bob-${uuid.v4()}`
+
+let aliceKey, bobKey, otherKey
+
+describe('Accounts test', () => {
+
+  describe('Promise style', () => {
+
+    before('set up API objects', () => {
+
+      // Key creation and signer setup
+      return expect(Promise.all([
+        client.mockHsm.keys.create({alias: aliceAlias}),
+        client.mockHsm.keys.create({alias: bobAlias}),
+        client.mockHsm.keys.create()
+      ])).to.be.fulfilled
+
+      .then(keys => {
+        aliceKey = keys[0]
+        bobKey = keys[1]
+        otherKey = keys[2]
+
+        signer.addKey(aliceKey, client.mockHsm.signerConnection)
+        signer.addKey(bobKey, client.mockHsm.signerConnection)
+      })
+    })
+
+    describe('Single account creation', () => {
+
+      it('creates two accounts', () => {
+        return expect(Promise.all([
+          client.accounts.create({alias: aliceAlias, rootXpubs: [aliceKey.xpub], quorum: 1}),
+          client.accounts.create({alias: bobAlias, rootXpubs: [bobKey.xpub], quorum: 1})
+        ])).to.be.fulfilled
```

Comment:
I think we can actually remove all of the `expected(...).to.be.fulfilled` from our entire suite. The main purpose of this was to have our single giant test print out a line number when at the Promise that failed instead of the end of the test.